### ISSUE TITLE
Panel is showing up as a new tab

### ISSUE THREAD

**Issue by adamreisnz** [steelbrain/linter-ui-default#405]
When clicking on the error summary:

![image](https://user-images.githubusercontent.com/490562/28950408-ce155746-7917-11e7-8965-95a20b821a38.png)

A new Linter tab opens up instead of showing the panel at the bottom of the file. This tab is empty (no errors show) even though it indicated an error was present:

![image](https://user-images.githubusercontent.com/490562/28950431-e34fd262-7917-11e7-8654-e0b0d3865d21.png)

Going to settings and selecting "Show Panel" when the Linter tab is closed will open the Linter tab again:

![image](https://user-images.githubusercontent.com/490562/28950443-01fea454-7918-11e7-8157-f52142e649ea.png)

Not sure what is going on, but it is very strange behaviour. I've updated Atom to the latest in master and updated all packages as well.

OS X 10.12.6
Atom 1.20.0-dev-577dfe8de
linter-ui-default 1.6.3

**Comment by steelbrain** [comment]
You can just drag the panel tab to the bottom dock and then it'll stay there and will open there next time

**Comment by adamreisnz** [comment]
Lol, horrible feature... 😧 

Dragging doesn't work. If I drag the tab, it just positions itself as a new tab in split screen:

![image](https://user-images.githubusercontent.com/490562/28950586-5688942a-7919-11e7-9463-1db7e3397740.png)

No idea where to drag it to for it to work.

Moreover, even without it being positioned properly, shouldn't the tab show the errors?

**Comment by adamreisnz** [comment]
Ok managed to reposition it with some fiddling, but man this is really an annoying feature.

I wish the Atom team would spend their time on fixing long standing issues or speeding up the editor instead of messing with things that weren't broken... :(

**Comment by steelbrain** [comment]
Unfortunately, can't help you there. You could try opening an issue on the Atom repo and ping Simurai and brainstorm for better UX, I might even jump in to voice my opinion.

**Comment by adamreisnz** [comment]
What about:

> Moreover, even without it being positioned properly, shouldn't the tab show the errors?

**Comment by steelbrain** [comment]
Whoops, skipped that part, check your `panelRepresents` config to see what's up with that, try changing it to Entire Project temporarily and see if something shows up

**Comment by adamreisnz** [comment]
Yes, only when it's changed to Entire Project does something show up in the tab panel.

While I understand how this might be nice for someone who always wants to see a linter overview with all project errors, when your panel is set to current file this panel will always be empty and that is very confusing.

Maybe you need to check if the panel is a tab, then always show Entire Project.

**Comment by Arcanemagus** [comment]
> when your panel is set to current file this panel will always be empty and that is very confusing.

When it is set to current file it, shockingly, should only be showing messages for the current file. Are you saying that there were messages for the current file and it wasn't showing them? Or that you want it to automatically switch to "Entire Project" if there are no messages for the current file?

**Comment by mehcode** [comment]
If the linter panel is being shown as a tab I imagine (unless panes are split) the current file is always null as it would be the linter label if shown.

**Comment by adamreisnz** [comment]
> When it is set to current file it, shockingly, should only be showing messages for the current file. Are you saying that there were messages for the current file and it wasn't showing them? Or that you want it to automatically switch to "Entire Project" if there are no messages for the current file?

When the setting is set to "current file", and the Linter panel is in "tab form", then it doesn't show any error results -- even if you click on the linter error summary in your file. 

The current file may be null in those circumstances, because understandably it's in it's own tab, and not linked to a file, but I think that in that case the linter tab should by default always show *all* errors from the project, and not rely on the user setting, because it doesn't make sense to show an empty tab, unless there are no errors.

So in summary:

- If linter is not in tab form, show errors either for file or for project, depending on setting
- If linter is in tab form, it can't show results for file, so always default to listing errors in project

Does that make sense?

**Comment by mehcode** [comment]
For the record, linter-ui-plus is project-wide by default with a toggle button (much like the toggle buttons on the find panel) to go to file mode (the latter doesn't exist yet but will when I get back to the project).

**Comment by adamreisnz** [comment]
You are talking about this setting yes?

![image](https://user-images.githubusercontent.com/490562/28991673-8be9b632-79df-11e7-843a-42c8102efa75.png)

I really don't want to see linter errors in the entire project, because that will drive me mad when I'm contracting for someone and working with a team where half of the people don't have lint on save enabled 😠

**Comment by mehcode** [comment]
@adamreisnz Actually no. That setting doesn't exist in [linter-ui-plus](https://github.com/mehcode/linter-ui-plus). 

The Panel in linter-ui-plus will have a button group like below where one of them means "current file".

![screen shot 2017-08-04 at 6 14 58 pm](https://user-images.githubusercontent.com/753919/28991685-d8c34a80-7940-11e7-8a91-83a8357713a1.png)

We should discuss this on a separate issue either in this or the linter-ui-plus repo, but I disagree on the project-wide only being a problem.

**Comment by steelbrain** [comment]
Whoops, I see the discussion has escalated a little so here's a little more explanation

When you drag the linter panel to the bottom, there's two possible scenarios where you could fit it

1. The bottom half of the screen as the pane
2. The bottom dock (by hovering over the middle of bottom bar and dropping on the ribbon that appears)

If you do the former, you won't ever be able to get the current file of errors to work, if you do the latter the current file scope should work perfectly fine. Hope that clears things up

**Comment by adamreisnz** [comment]
@steelbrain yes I am aware of this now, but do you reckon it makes sense?
If you have the setting enabled to show linter errors for the current file, and you go into tab form, you will never see a single error. I think this is highly confusing and people will not understand why they don't see errors. I didn't until you explained it.

So what I proposed earlier is for the UI to be "smart" about it:

- If linter is not in tab form, show errors either for file or for project, depending on setting
- If linter is in tab form, it can't show results for file, so always default to listing errors in project

**Comment by steelbrain** [comment]
I agree that it's confusing, maybe we could work it around by observing active panes and seeing if the current one is a linter one, if so, keep observing the last one.

Could you open a seperate issue for that as it is out of scope of this particular one. Thank you for being a linter user and your continued cooperation, appreciate it 👍
